### ISSUE TITLE
maybe endless loop in fileupload with firefox

### ISSUE THREAD

**Issue by aiomaster** [rack/rack#1308]
We have detected an issue using jquery file upload and a github version of rack.
Problem is, that larger fileuploads (larger than 50M) with firefox stop working in the rack middleware and produce 100% CPU load, so it looks like an infinit loop.
We have traced the commit that introduced the behaviour in rack: https://github.com/rack/rack/commit/534ef35bcc95b11719d2574a70757b61fa3eac33

The interesting stuff is, that the same files work in chrome but not in firefox.
So we compared the Request-Headers and the main difference is, that firefox uses other boundaries in the `Content-Type`-Header than Chrome.

Example-Header in Chrome:
`Content-Type: multipart/form-data; boundary=----WebKitFormBoundarydEXn76YwP9nKSKO3`

Example-Header in Firefox:
`Content-Type: multipart/form-data; boundary=---------------------------100673423314366417001080588648`

This is just a guessing, that the boundary parsing is involved here.

**Comment by aiomaster** [comment]
Ok it boils down to this line, that does not return somehow: https://github.com/rack/rack/blob/5559676e7b5a3107d39552285ce8b714b672bde6/lib/rack/multipart/parser.rb#L268

The content of the @body_regex is:
```
/(.*?)(
)?\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-84208446911303861091947234149(
|--)/m
```
I don't know if that helps in a way.

**Comment by jeremyevans** [comment]
Fixed by #1309
